clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -449,7 +449,11 @@ static bool shouldAssumeDSOLocal(const CIRGenModule &CGM,
     return false;
 
   if (CGOpts.DirectAccessExternalData) {
-    llvm_unreachable("-fdirect-access-external-data not supported");
+    if (auto gv = dyn_cast<mlir::cir::GlobalOp>(GV.getOperation()))
+      return !gv.getTlsModel().has_value();
```

Comment:
I don't think this is equivalent to the CodeGen code? That one only returns if the var isn't thread-local, but this one always returns. It's equivalent right now because of the way the function is laid out, but if things change in the future, it's probably easier to mirror the existing structure as closely as possible.

I'm linking the corresponding CodeGen code here for reference for everyone: https://github.com/llvm/llvm-project/blob/b2c615fc792c3f75af66aac99c05ffa85ef50354/clang/lib/CodeGen/CodeGenModule.cpp#L1726-L1744

Reply:
We should probably incorporate the OG cloning assumptions + asserts into the webpage docs, which I was never able to do, just in case you find yourself wanting to update some docs, this could be a good one!